pkg/controller/vitesscluster/reconcile_topo.go

```diff
@@ -85,43 +85,59 @@ func (r *ReconcileVitessCluster) reconcileCellTopology(ctx context.Context, vt *
 		desiredCells[cell.Name] = cell
 	}
 
-	// We need to add an alias for all the cells in each region so that vtgate
-	// knows that it can route traffic between them.
-	// Currently, only one region is supported so we allow routing anywhere.
-	//
-	// We also need to create the aliases before we create the cells because we
-	// don't want any vtgates to start after the cells are created but before
-	// the alias exists.
-	desiredCellsAliases := buildCellsAliases(desiredCells)
-	currentCellsAliases, err := ts.GetCellsAliases(ctx, true)
+	if *vt.Spec.TopologyReconciliation.RegisterCellsAliases {
+		err := r.registerCellsAliases(ctx, vt, ts, desiredCells)
+		if err != nil {
+			return resultBuilder.Error(err)
+		}
+	}
+
+	if *vt.Spec.TopologyReconciliation.RegisterCells {
+		result, err := r.registerCells(ctx, vt, ts, globalTopoImpl, desiredCells)
+		resultBuilder.Merge(result, err)
+	}
+
+	if *vt.Spec.TopologyReconciliation.PruneCells {
+		result, err := r.pruneCells(ctx, vt, ts, desiredCells)
+		if err != nil {
+			return resultBuilder.RequeueAfter(topoRequeueDelay)
+		} else {
+			resultBuilder.Merge(result, err)
```

Comment:
This can be brought out of the `else` block to reduce indentation of the happy path, since the preceding `if` follows the conventional pattern of `if error, return`.

Reply:
Done.